unittest/gunit/string-tests.cc

```diff
@@ -58,16 +58,8 @@ TEST(StringDeathTest, AppendEmptyString)
   tbl_name.append('.');
   tbl_name.append(String(table_name, system_charset_info));
   // We now have eight characters, c_ptr() is not safe.
-<<<<<<< HEAD
-#ifndef DBUG_OFF
-  MY_EXPECT_DEATH_IF_SUPPORTED(tbl_name.c_ptr(), ".*m_alloced_length >= .*");
-||||||| e5d189ecb94
-#ifndef DBUG_OFF
-  EXPECT_DEATH_IF_SUPPORTED(tbl_name.c_ptr(), ".*m_alloced_length >= .*");
-=======
 #ifndef NDEBUG
-  EXPECT_DEATH_IF_SUPPORTED(tbl_name.c_ptr(), ".*m_alloced_length >= .*");
->>>>>>> 37b047220a907c2a6d7235ddf2b7a6be916cc82e
+  MY_EXPECT_DEATH_IF_SUPPORTED(tbl_name.c_ptr(), ".*m_alloced_length >= .*");
```

Comment:
Please double-check if this works with both GCC and Clang Address Sanitizers

Reply:
This was checked with the fix from https://github.com/percona/percona-server/pull/4285/commits/66fa20ae1bc8ec7f92d98a88331bc5d7608a769e?

Reply:
Yes, it was. To be sure the address sanitizer flag had been applied, I run the test with the following env variable and checked a report was shown:

```
export ASAN_OPTIONS="atexit=1:...";
```